### PR TITLE
Gomod k8s v1.15.1

### LINKED ISSUE #748: Change to use gomod to manage the dependencies
**What would you like to be added**:
Use gomod to manage the dependies and remove the vendor from the project

**Why is this needed**:
To manage the dependiences more efficient

### PR DESCRIPTION
**Pull request by subpathdev**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
using gomod instead of gopkg

**Which issue(s) this PR fixes**:
Fixes #748

**Special notes for your reviewer**:
this pr based on #941 
@edisonxiang  is working together with me
Thanks @edisonxiang for the great work

### CODE REVIEW

**Comment by subpathdev** [comment]
/cc @edisonxiang @sids-b @kadisi @fisherxu @sujithsimon22 

this pr is a bit complicated; I use the master branch as base but I need the changes from #941 to build this successfully.
After merging the pr I have to fix the merge - conflicts (because removing Gopkg.toml / Gopkg.lock and vendor) => currently WIP;

**Comment by edisonxiang** [comment]
@subpathdev Please feel free to rebase it:) Thanks very much.

**Comment by subpathdev** [comment]
sry for the curious number of push force commits; I had struggle with building this on travis; When you use gomod and develop in you GOPATH please execute this: `export GO111MODULE=on`

**Comment by subpathdev** [comment]
The ci time increases with this pr because we don't use vendor any more. We improve the duration by using vendor :D (we have to change the build execution to `go build -mod vendor` and build the vendor directory by executing `go mod vendor`)

**Comment by subpathdev** [comment]
I have added the vendor directory to speed up the ci process;

Actual we can use local our own setup and the vendor directory is only used by github. (If you push a dependency change, you have to run `go mod vendor`)

**Comment by sids-b** [comment]
@subpathdev , can we keep the vendor files ? The code should be able to build after cloning
Even k8s upstream still keep the vendor after using go mod https://github.com/kubernetes/kubernetes

**Comment by sids-b** [comment]
@rohitsardesai83 @fisherxu thoughts ?

**Comment by subpathdev** [comment]
yes we can use the vendor directory. It can be build via `go mod vendor`. The default build will ignore the vendor file and still downloading the dependencies. (you can use `go build -mod=vendor` to force the usage of the vendor directory

**Comment by edisonxiang** [comment]
@subpathdev Thanks your work.
Could we keep vendor file folder in kubeeedge repostory like kubernetes guys did.
https://github.com/kubernetes/kubernetes/tree/master/vendor
So that we can track some changes for vendors:)
How about that?

**Comment by subpathdev** [comment]
I will add the vendor directory and use the vendor directory to build kubeedge

**Comment by fisherxu** [comment]
> I will add the vendor directory and use the vendor directory to build kubeedge

Thanks! @subpathdev We'd better keep the vendor for the code can be build.

**Comment by subpathdev** [comment]
I've readded  the vendor directory and build kubeedge with these files.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	k8s.io/gengo v0.0.0 => k8s.io/gengo v0.0.0-20190327210449-e17681d19d3a // indirect
+	k8s.io/heapster => k8s.io/heapster v1.2.0-beta.1 // indirect
+	k8s.io/klog => k8s.io/klog v0.3.3 // indirect
+	k8s.io/kube-aggregator v0.0.0 => k8s.io/kube-aggregator v0.0.0-20190718184434-a064d4d1ed7a
+	k8s.io/kube-controller-manager v0.0.0 => k8s.io/kube-controller-manager v0.0.0-20190718190030-ea930fedc880
+	k8s.io/kube-openapi v0.0.0 => k8s.io/kube-openapi v0.0.0-20190718094010-3cf2ea392886 // indirect
+	k8s.io/kube-proxy v0.0.0 => k8s.io/kube-proxy v0.0.0-20190718185641-5233cb7cb41e
+	k8s.io/kube-scheduler v0.0.0 => k8s.io/kube-scheduler v0.0.0-20190718185913-d5429d807831
+	k8s.io/kubelet v0.0.0 => k8s.io/kubelet v0.0.0-20190718185757-9b45f80d5747
+	//k8s.io/kubernetes v0.0.0 => k8s.io/kubernetes v1.15.1
```
Comment (rohitsardesai83) [inline go.mod]: is this needed ?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	k8s.io/csi-translation-lib v0.0.0 => k8s.io/csi-translation-lib v0.0.0-20190718190424-bef8d46b95de
+	k8s.io/gengo v0.0.0 => k8s.io/gengo v0.0.0-20190327210449-e17681d19d3a // indirect
+	k8s.io/heapster => k8s.io/heapster v1.2.0-beta.1 // indirect
+	k8s.io/klog => k8s.io/klog v0.3.3 // indirect
+	k8s.io/kube-aggregator v0.0.0 => k8s.io/kube-aggregator v0.0.0-20190718184434-a064d4d1ed7a
+	k8s.io/kube-controller-manager v0.0.0 => k8s.io/kube-controller-manager v0.0.0-20190718190030-ea930fedc880
+	k8s.io/kube-openapi v0.0.0 => k8s.io/kube-openapi v0.0.0-20190718094010-3cf2ea392886 // indirect
+	k8s.io/kube-proxy v0.0.0 => k8s.io/kube-proxy v0.0.0-20190718185641-5233cb7cb41e
+	k8s.io/kube-scheduler v0.0.0 => k8s.io/kube-scheduler v0.0.0-20190718185913-d5429d807831
+	k8s.io/kubelet v0.0.0 => k8s.io/kubelet v0.0.0-20190718185757-9b45f80d5747
```
Comment (rohitsardesai83) [inline go.mod]: cant we have release sha's for dependencies ?

File: go.mod (hunk 1 above)
Comment (subpathdev) [reply go.mod]: this is not needed => I will remove

File: go.mod (hunk 2 above)
Comment (subpathdev) [reply go.mod]: go mod can use two different types of dependency versions:
1. vx.y.z where x,y,z, is an integer value
2. v0.0.0-timestamp-<first 12 characters of the commit hash>

But I think I didn't get your point can you explain me what do you mean with sha's (

**Comment by sids-b** [comment]
We are removing a lot of code here... That is a very good thing.

Ping @edisonxiang @qizha @kevin-wangzefeng @fisherxu for review

**Comment by subpathdev** [comment]
remove merge conflicts

**Comment by sids-b** [comment]
@subpathdev can you please resolve the conflicts ?

**Comment by sids-b** [comment]
@kevin-wangzefeng @fisherxu can you please review the PR once before merge ?

**Comment by fisherxu** [comment]
> @kevin-wangzefeng @fisherxu can you please review the PR once before merge ?

yes, will take a look soon

**Comment by sids-b** [comment]
Thank you very much @subpathdev for the great efforts :) 

/approve

**Pull request merged**